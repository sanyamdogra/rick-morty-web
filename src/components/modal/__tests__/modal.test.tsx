import React from "react";
import { render, cleanup, fireEvent, screen } from "@testing-library/react";
import Modal from "..";

afterEach(cleanup);

test("clicking on cross calls the handleClose", () => {
  const handleClose = jest.fn();

  const { asFragment } = render(
    <Modal show={true} handleClose={handleClose}>
      <div>Test Content</div>
    </Modal>
  );

  expect(asFragment()).toMatchSnapshot();

  fireEvent.click(screen.getByTestId("modal-close-btn"));

  expect(handleClose).toHaveBeenCalledTimes(1);
});

test("setting the show to false should add --none class", () => {
  const handleClose = jest.fn();

  render(
    <Modal show={false} handleClose={handleClose}>
      <div>Test Content</div>
    </Modal>
  );

  expect(screen.getByTestId("modal-body")).toHaveAttribute(
    "class",
    "rick-morty-modal rick-morty-modal--none"
  );
});
