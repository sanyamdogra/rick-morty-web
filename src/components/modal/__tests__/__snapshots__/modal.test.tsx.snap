// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clicking on cross calls the handleClose 1`] = `
<DocumentFragment>
  <div
    class="rick-morty-modal rick-morty-modal--block"
    data-testid="modal-body"
  >
    <section
      class="rick-morty-modal__main"
    >
      <div>
        Test Content
      </div>
      <svg
        class="rick-morty-modal__close"
        data-testid="modal-close-btn"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 352 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        />
      </svg>
    </section>
  </div>
</DocumentFragment>
`;
